### ISSUE TITLE
[FEATURE] Add metric for last backup of a volume

### ISSUE THREAD

**Issue by koeberlue** [longhorn/longhorn#6049]
## Is your feature request related to a problem? Please describe (👍 if you like this request)

I am running the [`kube-prometheus-stack`](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) to monitor some kubernetes clusters. I have set up some alerts according to the [documentation](https://longhorn.io/docs/1.4.2/monitoring/alert-rules-example/). 

I want to configure an additional alert for any volume, where the backup is older than one week. In the UI I can see `Last Backup At` for every volume, but I can't scrape this value from the metrics. 

## Describe the solution you'd like

I'd like to have an additional metric, e.g. `longhorn_volume_last_backup_at` for each volume which either contains the timestamp of the last successful backup or some nil value if no backup was ever completed.

## Describe alternatives you've considered

I already built an alert using the available `longhorn_backup_state` metric which really helps to find failed or stuck backup jobs. But it only works for backup jobs that are already configured and I found it kind of hard to parse the date of the last successful backup. It also did not alert me for backups where I missed to enable backups.

**Comment by ejweber** [comment]
Thanks for opening this issue @koeberlue. I'm copying @PhanLe1010 and @derekbit on this one, since it looks like they've done the most work on our metrics collection. If they don't get a chance to get back to you, however, I'll look into this a bit too.

If you're familiar with Prometheus and know some Go, I'd encourage you to consider putting together a PR as well. We're always happy to have community contributions!

**Comment by koeberlue** [comment]
> If you're familiar with Prometheus and know some Go, I'd encourage you to consider putting together a PR as well. We're always happy to have community contributions!

Thanks for the encouraging words!

I'd consider myself a newbie when it comes to Prometheus and Go, but I gave it a try.

**Comment by innobead** [comment]
@c3y1huang Please help with the issue lifecycle, and also update the doc. Thanks.

**Comment by koeberlue** [comment]
I'm really sorry but I will not find the time to complete this PR. That's why I will close the PR for now. Anyone interested in completing this may feel free to pick up my code changes. Again, I am really sorry, but I will not find the time to continue working on this for now.

**Comment by alexander-bauer** [comment]
Hi y'all, I landed here after some searching, because I'm also looking to configure the same alerting. This would be huge for me to get alerted to gaps in my backup schedules / configuration.

It looks like this feature request was closed as _completed_, but it's not clear that it actually was, since the PR wasn't merged. I want to check and see if this is still on the roadmap, or whether you're open to more contributions here.

**Comment by c3y1huang** [comment]
> Hi y'all, I landed here after some searching, because I'm also looking to configure the same alerting. This would be huge for me to get alerted to gaps in my backup schedules / configuration.
>
> It looks like this feature request was closed as completed, but it's not clear that it actually was, since the PR wasn't merged. I want to check and see if this is still on the roadmap, or whether you're open to more contributions here.

@alexander-bauer , we are open to contributions! Unfortunately, @koeberlue was unable to complete it. I encourage you to create a PR for this.

https://github.com/longhorn/longhorn-manager/pull/1974

**Comment by derekbit** [comment]
@inqode-lars Can we assign the ticket to you because you are working on the ticket? Thanks.

**Comment by derekbit** [comment]
~~@inqode-lars Can you update the official document? Thanks.~~ @c3y1huang Thanks 
https://longhorn.io/docs/1.11.0/monitoring/metrics/
https://github.com/longhorn/website/blob/master/content/docs/1.11.0/monitoring/metrics.md

cc @sushant-suse

**Comment by c3y1huang** [comment]
> [@inqode-lars](https://github.com/inqode-lars) Can you update the official document? Thanks. [longhorn.io/docs/1.11.0/monitoring/metrics](https://longhorn.io/docs/1.11.0/monitoring/metrics/) [longhorn/website@`master`/content/docs/1.11.0/monitoring/metrics.md](https://github.com/longhorn/website/blob/master/content/docs/1.11.0/monitoring/metrics.md)
> 
> cc [@sushant-suse](https://github.com/sushant-suse)

https://github.com/longhorn/website/pull/1272

**Comment by inqode-lars** [comment]
> [@inqode-lars](https://github.com/inqode-lars) Can we assign the ticket to you because you are working on the ticket? Thanks.

Is there anything left to do? Or can this issue be closed?

**Comment by derekbit** [comment]
> > [@inqode-lars](https://github.com/inqode-lars) Can we assign the ticket to you because you are working on the ticket? Thanks.
> 
> Is there anything left to do? Or can this issue be closed?

No, eveything looks good.
QA will help verify it and close the ticket. Thank you.

**Comment by roger-ryao** [comment]
**Verified on master-head 20251229**

- longhorn master-head ([02d718](https://github.com/longhorn/longhorn/commit/02d71875e35c076b755bea569a2593304047e338)) 
- longhorn-manager master-head ([857db7](https://github.com/longhorn/longhorn-manager/commit/857db75fdef5e4de5093b593002beadce92bd4e3))

**The test steps**
1. Create a volume backup

2. Use the `curl` command to fetch metrics data from `longhorn-backend`
    ```bash
    > kl get service
    NAME                                       TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
    longhorn-admission-webhook                 ClusterIP   10.43.191.158   <none>        9502/TCP         3h34m
    longhorn-backend                           ClusterIP   10.43.192.15    <none>        9500/TCP         3h34m
    longhorn-frontend                          ClusterIP   10.43.146.234   <none>        80/TCP           3h34m
    longhorn-recovery-backend                  ClusterIP   10.43.233.131   <none>        9503/TCP         3h34m
    longhorn-ui-nodeport                       NodePort    10.43.34.24     <none>        8000:30000/TCP   3h32m
    pvc-7d4327b7-e816-4219-ab25-36d76ddd88c6   ClusterIP   10.43.195.223   <none>        2049/TCP         76m
    pvc-c421fb89-4e2e-45b7-90d2-19b9e23ea8e6   ClusterIP   10.43.32.230    <none>        2049/TCP         60m
    ```

3. ssh to control node
    Execute the command
    ```bash
    > curl -sSL http://10.43.192.15:9500/metrics | grep longhorn_volume_last_backup_at
    # HELP longhorn_volume_last_backup_at Unix timestamp of the last successful backup of this volume, or 0 if no such backup exists
    # TYPE longhorn_volume_last_backup_at gauge
    longhorn_volume_last_backup_at{node="ip-10-0-1-245",pvc="www-web-state-rwx-v1-0",pvc_namespace="default",    volume="pvc-7d4327b7-e816-4219-ab25-36d76ddd88c6"} 1.766991604e+09
    ```

**Result Passed**

1. After executing the CLI, we can retrieve volume metrics, including:
  
  - `longhorn_volume_last_backup_at`
